### ISSUE TITLE
Spring Security ACL

### ISSUE THREAD

**Issue by rafaelrenanpacheco** [eugenp/tutorials#9140]
Hi there! I followed the instructions on https://www.baeldung.com/spring-security-acl, but the `MethodSecurityExpressionHandler` configuration didn't work on Spring Boot 2.2.6. What worked was creating a bean of `AclPermissionEvaluator`. So, instead of:

```java
@Configuration
@EnableGlobalMethodSecurity(prePostEnabled = true, securedEnabled = true)
public class AclMethodSecurityConfiguration 
  extends GlobalMethodSecurityConfiguration {
 
    @Autowired
    MethodSecurityExpressionHandler 
      defaultMethodSecurityExpressionHandler;
 
    @Override
    protected MethodSecurityExpressionHandler createExpressionHandler() {
        return defaultMethodSecurityExpressionHandler;
    }
}

@Bean
public MethodSecurityExpressionHandler 
  defaultMethodSecurityExpressionHandler() {
    DefaultMethodSecurityExpressionHandler expressionHandler
      = new DefaultMethodSecurityExpressionHandler();
    AclPermissionEvaluator permissionEvaluator 
      = new AclPermissionEvaluator(aclService());
    expressionHandler.setPermissionEvaluator(permissionEvaluator);
    return expressionHandler;
}
```

I did:

```java
@Configuration
@EnableGlobalMethodSecurity(prePostEnabled = true, securedEnabled = true)
public class MethodSecurityConfiguration extends GlobalMethodSecurityConfiguration {
}

@Configuration
public class AclConfiguration {

    @Bean
    public AclPermissionEvaluator aclPermissionEvaluator() {
        return new AclPermissionEvaluator(aclService());
    }

    /* Other ACL configurations */
}
```

**Comment by kwoyke** [comment]
Hi @rafaelrenanpacheco,
Thanks for the feedback! I'm sorry to hear you face issues with this.
We'll take a look into this. But can you please clarify when does the problem occurs? Is it during startup? Is any exception thrown? Or does some functionality doesn't work as expected?
I'd appreciate your input here.

**Comment by rafaelrenanpacheco** [comment]
Hello!

With the first configuration the ACL was just ignored, with the second configuration it worked as expected, and ACL policy was applied when calling secured methods.

I'm no longer using Spring ACL, so if this issue doesn't happen on your application, I think this issue may be closed.

Best regards,
Rafael Pacheco.

**Comment by kwoyke** [comment]
Thanks for the reply @rafaelrenanpacheco!
Well, I've just checked our example project and the ACL is working fine.
Just to clarify, have you observed the issue mentioned above in our sample application? Or was it one that you've created yourself?

**Comment by rafaelrenanpacheco** [comment]
It was on an application I created myself. I followed the ACL tutorial and at the end, when calling a HTTP endpoint that calls a secured method, the ACL was completely ignored. After removing the `MethodSecurityExpressionHandler` bean and creating a bean for `AclPermissionEvaluator` it started to work just fine. In this project I'm also using Spring Cloud Security, so it may have caused some configuration conflict. Also, I just saw that the spring project used for this tutorial is using the current version of Spring Boot, and, if it is working as you said, I think we can close this issue.

Best regards,
Rafael Pacheco.
